pkg/controller/volume/cache/desired_state_of_world.go

```diff
@@ -57,7 +57,7 @@ type DesiredStateOfWorld interface {
 	// should be attached to the specified node, the volume is implicitly added.
 	// If no node with the name nodeName exists in list of nodes managed by the
 	// attach/detach attached controller, an error is returned.
-	AddPod(podName types.UniquePodName, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
+	AddPod(pod *api.Pod, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
```

Comment:
Leave `podName types.UniquePodName` as is. It should be the key to the maps. If you need additional information, like podName/podNamespace/podUID, pass them in as additional parameters (just the fields instead of the entire giant pod object), and save them in the internal `pod` struct for use later.